File: web/src/mock/TobePurchaseBom.js
import Mock from 'mockjs'

Mock.setup({
  timeout: '100-1000'
})

Mock.mock(RegExp('^(/Data/TobePurchaseBom)' + '.*'), options => {
  // list = {data:[{},{}]}
  const list = Mock.mock({
    'data|1-100': [
      {
        ProjectNO: /(Z|H|S|T|M)CC\d{6}-\d{1,2}/,
        OrderType: /(NOR|REP|PRS|RCL)/,
        WorkOrderNO: /(Z|H|S|T|M)OF\d{8}/,
        'BomSeq|1-100': 1,
        CustomerCode: /0000\d{1}/,
        CustomerName: () => { return Mock.mock('@sentence(3, 5)') },
        ForPN: /([A-Z]){6}/,
        PN: /([A-Z]){6}/,
        Description: () => { return Mock.mock('@sentence(3, 5)') },
        'Qty|1-100': 1,
        'ShortQty|1-100': 1,
        'AllQty|1-100': 1,
        Unit: 'UN',
        CreateDate: () => { return Mock.mock('@date("yyyy-MM-dd")') }
      }
    ]
  })
  return list.data
})

File: data-srv/src/main/java/com/da/sage/assistant/model/TobePurchaseBom.java
/*********************************************************************************************************************
 * @Author                : Robert Huang<56649783@qq.com>                                                            *
 * @CreatedDate           : 2022-03-26 17:01:00                                                                      *
 * @LastEditors           : Robert Huang<56649783@qq.com>                                                            *
 * @LastEditDate          : 2023-03-12 13:22:36                                                                      *
 * @FilePath              : src/main/java/sageassistant/dataSrv/model/TobePurchaseBom.java                           *
 * @CopyRight             : Dedienne Aerospace China ZhuHai                                                          *
 ********************************************************************************************************************/

package com.da.sage.assistant.model;

import com.da.sage.assistant.model.base.ModelTemplate;
import java.util.Date;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class TobePurchaseBom extends ModelTemplate {

    private static final long serialVersionUID = 1L;

    private String ProjectNO;
    private String OrderType;
    private String WorkOrderNO;
    private String BomSeq;
    private String CustomerCode;
    private String CustomerName;
    private String ForPN;
    private String PN;
    private String Description;
    private Integer Qty;
    private Integer ShortQty;
    private Integer AllQty;
    private String Unit;
    private Date CreateDate;
    private String CreateUser;
}
